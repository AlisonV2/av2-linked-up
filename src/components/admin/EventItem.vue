<template>
  <div class="card text-center">
    <div class="card-header title">
      {{ event.name }}
    </div>
    <p class="card-text">{{ event.description }}</p>
    <p class="card-text">{{ event.localisation }}</p>
    <div class="card-footer text-muted">
      {{ event.date }}
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports EventItem
 * @type {Component}
 * @vue-prop {object} event - event object
 * @vue-data {string} role - User role
 */
export default {
  name: 'eventItem',
  props: ['event'],
  data() {
    return {
      role: '',
    };
  },
  created() {
    this.$store
      .dispatch('getUserRole')
      .then(() => {
        this.role = this.$store.getters.getUserRole;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
