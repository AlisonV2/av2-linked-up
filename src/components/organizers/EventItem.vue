<template>
  <div class="card Event-card">
    <div class="card-overlay"></div>
    <div class="card-details fade-in-top text-center">
      <h5 class="card-title" :data-test="event.id">{{ event.name }}</h5>
      <p class="card-text">{{ event.city }} ({{ event.zip }})</p>
    </div>
    <img
      :src="event.thumbnail"
      class="card-img-top"
      :alt="event.id"
    />
    <div class="card-body">
      <h5 class="card-title">{{ event.name }}</h5>
    </div>
  </div>
</template>

<script>
/**
 * @exports EventItem
 * @type {Component}
 * @vue-prop {object} events
 */
export default {
  name: 'EventItem',
  props: ['event'],
};
</script>

<style lang="scss" scoped>
.event-card {
  .card-title {
    font-family: $accent-font;
    color: $primary;
    font-size: 1.2rem;
  }
  .card-text {
    color: $light;
  }
  .card-details {
    position: absolute;
    text-align: center;
    padding-left: 1em;
    padding-right: 1em;
    width: 100%;
    top: 50%;
    left: 50%;
    opacity: 0;
    transform: translate(-50%, -50%);
    transition: all 0.3s ease-in-out 0s;
    .card-title {
      margin-bottom: 2rem;
    }
  }
  .card-overlay {
    background: rgba(0, 0, 0, 0.8);
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    opacity: 0;
    transition: all 0.4s ease-in-out 0s;
  }
}

.Event-card:hover {
  .card-overlay {
    opacity: 1;
  }
  .card-details {
    display: block;
    top: 50%;
    left: 50%;
    opacity: 1;
  }
}

.fade-in-top {top: 20%};
</style>
