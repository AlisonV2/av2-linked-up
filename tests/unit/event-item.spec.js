import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import EventItem from '@/components/admin/EventItem.vue';

const actions = {
  getUserRole: jest.fn(),
};

const getters = {
  getUserRole: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return {
      role: 'organizer',
      event: { 
          name: '',
          orgaName: '',
          description: '',
          localisation: '',
          date: ''
      }
    };
  },
  global: {
    plugins: [store],
  },
};

/**
 * @module EventItemTest
 */
describe('EventItem.vue', () => {
  it('getUserRole method should be called', async () => {
    const component = shallowMount(EventItem, config);
    EventItem.created.call(component.vm);
    await component.vm.$nextTick();
    expect(component.vm.getUserRole).toHaveBeenCalled;
  });

  it('getUserRole action should be called', async () => {
    const component = shallowMount(EventItem, config);
    EventItem.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getUserRole).toHaveBeenCalled;
  });

  it('getUserRole getter should be called', async () => {
    const component = shallowMount(EventItem, config);
    EventItem.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getUserRole).toHaveBeenCalled;
  });
});
