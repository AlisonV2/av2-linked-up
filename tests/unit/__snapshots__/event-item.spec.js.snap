// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventItem.vue EventItem Snapshot 1`] = `
<div
  class="card event-card"
>
  <div
    class="card-overlay"
  />
  <div
    class="card-details fade-in-top text-center"
  >
    <h5
      class="card-title"
      data-test="gTvQIqYFzXCLID4TIU5u"
    >
      Event
    </h5>
    <p
      class="card-text"
    >
      BÃ¨gles (33130)
    </p>
  </div>
  <img
    alt="gTvQIqYFzXCLID4TIU5u"
    class="card-img-top"
    src="https://via.placeholder.com/150"
  />
  <div
    class="card-body"
  >
    <h5
      class="card-title"
    >
      Event
    </h5>
  </div>
</div>
`;
